File: prob2/prob2.py
from random import randrange
from math import sqrt
 
def rand_n(n):
    lst = []
    for i in xrange(n):
	lst.append(randrange(100))
    return lst
 
def average(lst):
    avg = sum(lst) / len(lst)
    return avg
 
def variance(lst,avg):
    var_list = []
    for num in lst:
	var_list.append((num-avg)**2)
    variance = sum(var_list) / len(var_list)
    return variance
 
def std_dev(variance):
    std = sqrt(variance)
    return std
 
lst = rand_n(10)
print "The numbers are " + str(lst)[1:-1]
 
avg = average(lst)
print "The average is " + str(avg)
 
var = variance(lst,avg)
print "The variance is " + str(var)
 
std = std_dev(var)
print "The standard dev is " + str(round(std,2))

File: prob2/prob2.rb
def rand_n(n)
  list = []
  n.times do 
    list << rand(1..100)
  end
  list
end
 
def average(list)
  sum = list.inject{|sum,x| sum + x }
  avg = sum / list.length()
end
 
def variance(list,average)
  var_list =[]
  list.each do |num|
    var_list << ((num - average)**2)
  end
  variance = var_list.inject{|sum,x| sum + x } / list.length()
end
 
def std_dev(variance)
  standard_dev = Math.sqrt(variance)
end
 
test = rand_n(10)
puts "The numbers are:"
test.each do |num|
  print num.to_s + ", "
end

avgerage = average(test)
puts "\nThe average is " + avgerage.to_s

var = variance(test,avgerage)
puts "The variance is " + var.to_s

std = std_dev(var)
puts "The standard dev is " + (std).round.to_s